src/nss-resolve/nss-resolve.c

```diff
@@ -339,7 +314,7 @@ enum nss_status _nss_resolve_gethostbyname3_r(
 
         r = sd_bus_open_system(&bus);
         if (r < 0)
-                goto fallback;
+                return NSS_STATUS_UNAVAIL;
```

Comment:
this doesn't look right... you need go go through the "fail" label here, as we really should initialize the two other error parameters on failure too.